### PR TITLE
Feature/adding payment info update changes

### PR DESCRIPTION
**Pull request by fmmajd**
This pull request will:
* add paymentInfo to cash payback ednpoint
* add an endpoint to update payment-info records

### CODE REVIEW

**Review by parisaam — APPROVED** [review]

File: src/Treasury/Communicator.php [hunk 1]
```diff
@@ -112,23 +101,22 @@ public function storeCreditPayBack(CreditPayBackDto $creditPayBackDto)
...
             'value' => $creditPayBackDto->getValue(),
             'creator_id' => $creditPayBackDto->getCreatorId(),
+            'payment_info_id' => $creditPayBackDto->getPaymentInfoId(),
         ]);
         return $response->getStatusCode() == 200;
     }
 
     public function storeOrderPayment(OrderPayDto $orderPayDto): bool
     {
         $uri      = "api/v1/orders/{$orderPayDto->getOrderId()}/payments/{$orderPayDto->getPaymentId()}";
```
Comment (aimanizadeh) [inline src/Treasury/Communicator.php:111]: We can use function for removing URI prefix

File: src/Treasury/Communicator.php [hunk 2]
```diff
@@ -112,23 +101,22 @@ public function storeCreditPayBack(CreditPayBackDto $creditPayBackDto)
         $response = $this->request(static::METHOD_POST, $uri, [
             'value' => $creditPayBackDto->getValue(),
             'creator_id' => $creditPayBackDto->getCreatorId(),
+            'payment_info_id' => $creditPayBackDto->getPaymentInfoId(),
         ]);
         return $response->getStatusCode() == 200;
```
Comment (aimanizadeh) [inline src/Treasury/Communicator.php:106]: Add blank line before return for better readability

**Review by miladkiaei — APPROVED** [review]

File: src/Treasury/Dto/PaymentInfoUpdateDto.php [hunk 3]
```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace SnappMarket\Treasury\Dto;
+
+use Exception;
+
+class PaymentInfoUpdateDto
+{
+    const ACTION_PAID = 'paid';
```
Comment (aimanizadeh) [inline src/Treasury/Dto/PaymentInfoUpdateDto.php:9]: I think you can use Enum for defining types

File: src/Treasury/Dto/PaymentInfoUpdateDto.php [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+    public function setValue(int $value): void
+    {
+        $this->value = $value;
+    }
+
+    public function setAction(string $action): void
+    {
+        if (!in_array($action, static::ACCEPTABLE_ACTIONS)) {
+            throw new Exception("Action is not valid.");
```
Comment (aimanizadeh) [inline src/Treasury/Dto/PaymentInfoUpdateDto.php:61]: You can put a @throws entry in the docblock

**Pull request merged**